### ISSUE TITLE
Recursive use of "Not" rule do not display the proper message

### ISSUE THREAD

**Issue by Rafael-BP** [Respect/Validation#479]
``` phpt
--FILE--
<?php
require 'vendor/autoload.php';

use Respect\Validation\Validator as v;
use Respect\Validation\Exceptions\OptionalException;
use Respect\Validation\Exceptions\AllOfException;

try {
    v::not(v::not(v::optional(v::equals('foo'))->setName('My field')))->check('foo');
} catch (OptionalException $e) {
    echo $e->getMainMessage().PHP_EOL;
}
?>
--EXPECTF--
My field must be optional
```

Test not working,
I talked to @henriquemoody and the exception of OptionalException don't trigger.

**Comment by henriquemoody** [comment]
I will try to fix for this version but I'm pretty sure it's not possible to fix it using the current codebase. Thanks for reporting, BTW.

**Comment by batusa** [comment]
```php
v::not(v::not(v::optional(v::equals('foo'))->setName('My field')))->check('foo');
```
why does it should throw an exception? I think it should throw an exception only in case the value passed in `check` were not: *empty string, null or "foo"* (I may be wrong).

I tested it here and it is returning true. the weird thing is when I pass a different value (for example `"a"`) and it throws a `NotException` with an "incomplete" message, using `getFullMessage()`, that's what I get:
```
\-These rules must not pass for "a"
```
but it doesn't say which rules.

**Comment by henriquemoody** [comment]
I'm closing very old issues.

Feel free to create another one if you find this is still relevant! 

Thanks for reporting! 🐼
